segmentation/utils/spatial_analysis_utils.py

```diff
@@ -250,3 +250,55 @@ def calculate_enrichment_stats(close_num, close_num_rand):
     dims = ["stats", "rows", "cols"]
     stats_xr = xr.DataArray(stats_data, coords=coords, dims=dims)
     return stats_xr
+
+
+def compute_neighbor_counts(fov_data, dist_matrix, distlim, pheno_num,
+                            cell_neighbor_counts, cell_neighbor_freqs, cell_count):
+    """Calculates the number of neighbor phenotypes for each cell
+
+    Args:
+        fov_data: data for the current fov
+        dist_matrix: cells x cells matrix with the euclidian
+            distance between centers of corresponding cells
+        distlim: threshold for spatial enrichment distance proximity
+        pheno_num: number of different cell phenotypes
+        cell_neighbor_counts: matrix with phenotype counts per cell
+        cell_neighbor_freqs: matrix with phenotype frequencies of
+            counts per phenotype/total phenotypes for each cell
+        cell_count: current cell in analysis
+    Returns:
+        cell_neighbor_counts: matrix with phenotype counts per cell
+        cell_neighbor_freqs: matrix with phenotype frequencies of
+            counts per phenotype/total phenotypes for each cell
+        cell_count: current cell in analysis"""
+
+    cell_label_col = "cellLabelInImage"
+
+    for j in list(fov_data.index):
+        # get specific cell
+        cell = fov_data.loc[j, cell_label_col]
+        # get all cell neighbors within threshold distance
+        cell_dist_mat = dist_matrix[int(cell - 1), :]
```

Comment:
I'm pretty sure the variable "cell" is meant to always be at least 1, but it never hurts to have a check to make sure we're not accidentally passing in negative values which could lead to unintended behavior.